File: days/01/main.go
package main

import (
	"fmt"
	"os"
)

func main() {
	if len(os.Args) < 2 {
		fmt.Println("Usage: main.go <input_file>")
		os.Exit(1)
	}

	inputFile := os.Args[1]
	input, err := os.ReadFile(inputFile)
	if err != nil {
		fmt.Printf("Error reading file: %v\n", err)
		os.Exit(1)
	}

	fmt.Printf("Input: %s\n", input)
}

File: days/01/main.zig
const std = @import("std");

pub fn main() !void {
    var args = std.process.args();
    _ = args.skip();
    const input_file = args.next() orelse return error.MissingArgument;

    const file = try std.fs.cwd().openFile(input_file, .{});
    defer file.close();

    const input = try file.readToEndAlloc(std.heap.page_allocator, std.math.maxInt(usize));
    defer std.heap.page_allocator.free(input);

    std.debug.print("Input: {s}\n", .{input});
}
